src/pykeen/models/base.py

```diff
@@ -1195,3 +1197,174 @@ def _can_slice(fn) -> bool:
 
 class MultimodalModel(EntityRelationEmbeddingModel):
     """A multimodal KGE model."""
+
+
+class InteractionFunction(nn.Module):
+    """Base class for interaction functions."""
```

Comment:
what do you think about the InteractionFunction having access to the model? This way, some of the parameters and embeddings can be used by the functions without having to do a lot of passing around

Reply:
Keeping it functional eases testing and results in a clearer data flow.

Reply:
Also it makes it easier to use pykeen interaction functions outside of pykeen, and to use interaction functions from the outside with pykeen.

Reply:
You’ve got a good point. However, I was trying to come up with a way to use this abstract for TransD, whose interaction function needs the embeddings and the indexes to make projections, and it wasn’t fitting so well

Reply:
Extend the interaction function interface to `(h, r, t, **kwargs)`?

Reply:
Maybe for this PR we should just focus on the simplest case, where the interaction function only needs (h,r,t) embeddings. Then in a follow-up PR, we can create a new abstraction that can handle more abstract operations

Reply:
Too late, I added ConvE :sweat_smile: